core/types/rollup_cost.go

```diff
@@ -283,3 +393,105 @@ func l1CostHelper(gasWithOverhead, l1BaseFee, scalar *big.Int) *big.Int {
 	fee.Mul(fee, l1BaseFee).Mul(fee, scalar).Div(fee, oneMillion)
 	return fee
 }
+
+func extractEcotoneFeeParams(l1FeeParams []byte) (l1BaseFeeScalar, l1BlobBaseFeeScalar *big.Int) {
+	offset := scalarSectionStart
+	l1BaseFeeScalar = new(big.Int).SetBytes(l1FeeParams[offset : offset+4])
+	l1BlobBaseFeeScalar = new(big.Int).SetBytes(l1FeeParams[offset+4 : offset+8])
+	return
+}
+
+func extractFjordFeeParams(l1FeeParams []byte) (l1CostIntercept, l1CostFastlzCoef, l1CostTxSizeCoef *big.Int) {
+	offset := fjordSectionStart
+	l1CostIntercept = big.NewInt(int64(bytesToInt32(l1FeeParams, offset, offset+4)))
+	l1CostFastlzCoef = big.NewInt(int64(bytesToInt32(l1FeeParams, offset+4, offset+8)))
+	l1CostTxSizeCoef = big.NewInt(int64(bytesToInt32(l1FeeParams, offset+8, offset+12)))
+	return
+}
+
+func bytesToInt32(data []byte, offsite, size int) int32 {
+	return int32(binary.BigEndian.Uint32(data[offsite : offsite+size]))
+}
+
+func bedrockCalldataGasUsed(costData RollupCostData) (calldataGasUsed *big.Int) {
+	calldataGas := (costData.zeroes * params.TxDataZeroGas) + (costData.ones * params.TxDataNonZeroGasEIP2028)
+	return new(big.Int).SetUint64(calldataGas)
+}
+
+// FlzCompressLen returns the length of the data after compression through FastLZ, based on
+// https://github.com/Vectorized/solady/blob/5315d937d79b335c668896d7533ac603adac5315/js/solady.js
+func FlzCompressLen(ib []byte) uint32 {
```

Comment:
Can we add a fuzz test to this?  The version in the compression analysis script panics with short inputs for example.